app/script_variant.py

```diff
@@ -65,6 +69,51 @@ def romanize_japanese_name_by_name_dict(word):
     return [word]
 
 
+def romanize_japanese_name_by_name_dict(word, for_index=True):
+    """
+    This method romanizes japanese name by using name dictionary.
+    If word isn't found in dictionary, this method doesn't
+    apply romanize.
+    This method can return multiple romanizations.
+    (because there are multiple ways to read the same kanji name in japanese)
+    Args:
+        for_index: this method is called for indexing or not
+                   when you want to romanize query txt, set this False
+    Returns:
+        [romanized_jp_name, and romanized_jp_name(split_word), ...]
+    """
+    if not word:
+        return ['']
+
+    words = []
+    for index in xrange(1, len(word)):
+        # split word because query word may not contains white space
+        # to support romanizing fullname without space.
+        # e.g., if the query (e.g., "山田太郎") doesn't contain white space,
+        # it would work return words ("yamatataro, yamadataro")
+        first_part = word[:index]
+        last_part = word[index:]
+        romanized_first_parts = romanize_single_japanese_word_by_name_dict(
+            first_part)
+        romanized_last_parts = romanize_single_japanese_word_by_name_dict(
+            last_part)
+        for romanized_first_part in romanized_first_parts:
+            for romanized_last_part in romanized_last_parts:
+                if romanized_first_part != first_part and \
+                        romanized_last_part != last_part:
+                    words.append(romanized_first_part+romanized_last_part)
+                    # if record contains a full name with kanji (e.g., "山田太郎"),
```

Comment:
Can you add an example of a set of the query and the record where you need these fields for indexing?

Also, can you add an example of a set of the query and the record where it would be problematic if you add these fields for query?